src/test/server/ServerStatus_test.cpp

```diff
@@ -19,164 +19,542 @@
+        std::string const& proto,
+        std::string const& user,
+        std::string const& password,
+        bool subobject = false)
+    {
+        Json::Value jrr;
+
+        Json::Value jp = Json::objectValue;
+        if(! user.empty())
+        {
+            jp["admin_user"] = user;
+            if(subobject)
+            {
+                //special case of bad password..passed as object
+                Json::Value jpi = Json::objectValue;
+                jpi["admin_password"] = password;
+                jp["admin_password"] = jpi;
+            }
+            else
+            {
+                jp["admin_password"] = password;
+            }
+        }
+
+        if(boost::starts_with(proto, "h"))
+        {
+            auto jrc = makeJSONRPCClient(env.app().config());
+            jrr = jrc->invoke("ledger_accept", jp);
+        }
+        else
+        {
+            auto wsc = makeWSClient(env.app().config(), proto == "ws2");
+            jrr = wsc->invoke("ledger_accept", jp);
+        }
+
+        return jrr;
+    }
+
+// ------------
+//  Test Cases
+// ------------
+    void
+    testWSClientToHttpServer(boost::asio::yield_context& yield)
+    {
+        testcase("WS client to http server fails");
+        using namespace jtx;
+        Env env(*this, []()
+            {
+                auto p = std::make_unique<Config>();
+                setupConfigForUnitTests(*p);
+                p->section("port_ws").set("protocol", "http,https");
+                return p;
+            }());
+
+        //non-secure request
+        {
+            boost::system::error_code ec;
+            beast::http::response_v1<beast::http::string_body> resp;
+            doWSRequest(env, yield, false, resp, ec);
             if(! BEAST_EXPECTS(! ec, ec.message()))
                 return;
             BEAST_EXPECT(resp.status == 401);
         }
 
-        // secure socket
+        //secure request
         {
-            ssl::context ctx{ssl::context::sslv23};
-            ctx.set_verify_mode(ssl::verify_none);
-            ssl::stream<ip::tcp::socket> ss{ios, ctx};
-            connect(ss.next_layer(), r.resolve(
-                ip::tcp::resolver::query{"127.0.0.1", to_string(*port)}));
-            ss.handshake(ssl::stream_base::client, ec);
+            boost::system::error_code ec;
+            beast::http::response_v1<beast::http::string_body> resp;
+            doWSRequest(env, yield, true, resp, ec);
             if(! BEAST_EXPECTS(! ec, ec.message()))
                 return;
-            write(ss, req, ec);
+            BEAST_EXPECT(resp.status == 401);
+        }
+    }
+
+    void
+    testStatusRequest(boost::asio::yield_context& yield)
+    {
+        testcase("Status request");
+        using namespace jtx;
+        Env env(*this, []()
+            {
+                auto p = std::make_unique<Config>();
+                setupConfigForUnitTests(*p);
+                p->section("port_rpc").set("protocol", "ws2,wss2");
```

Comment:
I did not change the config or intent from the existing test: https://github.com/ripple/rippled/blob/develop/src/test/server/ServerStatus_test.cpp#L115

I think that we are verifying that simple http GETs are still allowed by a websocket server because they just don't upgrade to WS.